wagtail/snippets/wagtail_hooks.py

```diff
@@ -49,10 +47,4 @@ def register_snippets_menu_item():
     )
 
 
-@hooks.register("register_permissions")
-def register_permissions():
-    content_types = ContentType.objects.get_for_models(*get_snippet_models()).values()
-    return Permission.objects.filter(content_type__in=content_types)
```

Comment:
Removing this in favour of `ModelViewSet`'s new built-in mechanism. This allows devs to customize which permissions get registered for snippets as well. Although it does mean we'll be doing a union for each model rather than a bulk query like we did here, and I'm not sure how much it affects the performance...

Alternatively we could override `register_permission()` in `SnippetViewSet` to be a no-op, and keep this here.

Reply:
I think this change is worth it for the extra flexibility and consistency between ModelViewSet and SnippetViewSet.

If performance is a concern, I think we could reasonably move this to a one-time query on first invocation of GroupForm rather than every instantiation of it - I don't think dynamically changing the permission set at runtime is something we need to support. (Doesn't have to happen here, though.)